src/main/java/com/keniu/services/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+package com.keniu.services;
+
+import com.keniu.dto.CreateCartItemRequestDto;
+import com.keniu.dto.ShoppingCartDto;
+import com.keniu.dto.UpdateCarItemRequestDto;
+import com.keniu.exceptions.EntityNotFoundException;
+import com.keniu.mappers.ShoppingCartMapper;
+import com.keniu.models.Book;
+import com.keniu.models.CartItem;
+import com.keniu.models.ShoppingCart;
+import com.keniu.repositories.ShoppingCartRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+
+    @Override
+    public ShoppingCart save(ShoppingCart shoppingCart) {
+        return shoppingCartRepository.save(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCartDto getShoppingCart(String email) {
+        ShoppingCart shoppingCart = getShoppingCartByEmail(email);
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCartDto addCartItem(String email,
+            CreateCartItemRequestDto createCartItemRequestDto) {
+        ShoppingCart shoppingCart = getShoppingCartByEmail(email);
+        CartItem cartItem = new CartItem();
+        cartItem.setShoppingCart(shoppingCart);
+        Book book = new Book();
+        book.setId(createCartItemRequestDto.getBookId());
```

Comment:
What if book that you are trying to add is not exists?

Reply:
We will get Exception:
Unable to find Book with id {}

Reply:
We’ve already discussed this a thousand times

Reply:
...